ecl/hql/hqlgram.y

```diff
@@ -2091,7 +2091,8 @@ conditionalAssignmentElseClause
                             parser->normalizeExpression($2);
                             parser->ensureBoolean($2);
                             //normalizeExpression($2, type_boolean, false);
-                            OwnedHqlExpr map = createValue(no_mapto, $2.getExpr(), $4.getExpr());
+                            OwnedHqlExpr other = $4.getExpr();
+                            OwnedHqlExpr map = createValue(no_mapto, other->getType(), $2.getExpr(), LINK(other));
```

Comment:
Previously, wouldn't the expression type be $2->getType()?

Reply:
Yes, and it was wrong.  The type of   expr1=>expr2 should be the type of expr2.